Add french Version

phrobinet opened the pull request:
<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe: 
Here is my French version of the documentation.

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch.
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
```
Hello, 

Here is my French version of the documentation.
Do not hesitate to make a return if you want modifications.

Have a nice day
Philippe

```

phrobinet committed:
add french version

phrobinet committed:
add more

lmiller1990 commented:
Wow, amazing! I will ping someone who knows French to give this a quick 👀

Akryum commented on docs/fr/README.md:
```diff
@@ -0,0 +1,90 @@
+
+En bref :
+
+- [Vue Test Utils 1](https://github.com/vuejs/vue-test-utils/) objectifs [Vue 2](https://github.com/vuejs/vue/).
```
```suggestion
- [Vue Test Utils 1](https://github.com/vuejs/vue-test-utils/) ciblant [Vue 2](https://github.com/vuejs/vue/).
```

Akryum commented on docs/fr/README.md:
```diff
@@ -0,0 +1,90 @@
+En bref :
+
+- [Vue Test Utils 1](https://github.com/vuejs/vue-test-utils/) objectifs [Vue 2](https://github.com/vuejs/vue/).
+- [Vue Test Utils 2](https://github.com/vuejs/vue-test-utils-next/) objectifs [Vue 3](https://github.com/vuejs/vue-next/).
```
```suggestion
- [Vue Test Utils 2](https://github.com/vuejs/vue-test-utils-next/) ciblant [Vue 3](https://github.com/vuejs/vue-next/).
```

Akryum commented on docs/fr/api/components/README--.md:
```diff
@@ -0,0 +1,4 @@
+# Components
+
+Vue Test Utils comprend des composants utilitaires que vous pouvez utiliser pour avoir des composants stubs
```
```suggestion
Vue Test Utils comprend des composants utilitaires que vous pouvez utiliser pour avoir des composants de remplacement (stubs).
```

cexbrayat commented on docs/fr/api/config.md:
```diff
@@ -0,0 +1,114 @@
+- type: `Boolean`
+- default: `true`
+
+Contrôler s'il faut ou non afficher des avertissements de dépréciation . Lorsqu'il est sur `true`, tous les avertissements de déprédation sont visibles dans la console.
```
déprédation -> déprécation

cexbrayat commented on docs/fr/api/config.md:
```diff
@@ -0,0 +1,114 @@
+- type: `Object`
+- default: `{}`
+
+Comme pour les `stubs`, les valeurs passées à `config.mocks` sont utilisées par défaut. Toute valeur passées à l'objet d'options de montage `mocks` aura la priorité sur celles déclarées dans `config.mocks`.
```
passées -> passée

cexbrayat commented on docs/fr/api/createLocalVue.md:
```diff
@@ -0,0 +1,59 @@
+
+- **Usage:**
+
+`createLocalVue` renvoie une classe Vue pour que vous puissiez ajouter des composants, des mixins et installer des plugins dans polluer la classe Vue globale.
```
dans -> sans

cexbrayat commented on docs/fr/api/render.md:
```diff
@@ -0,0 +1,103 @@
+
+Rend un objet en chaîne de caractères et retourne un [cheerio wrapper](https://github.com/cheeriojs/cheerio).
+
+Cheerio est une bibliothèque de type jQuery pour parcourir le DOM dans Node.js. Elle possède une API similaire à celles de [`Wrapper`](wrapper/) Vue Test Utils .
```
celles -> celle

cexbrayat commented on docs/fr/api/wrapper-array/README.md:
```diff
@@ -0,0 +1,29 @@
+
+### `length`
+
+`number` (lecture seulement): le nombre de `Wrappers`constenus dans le `WrapperArry`
```
constenus -> contenus (et espace manquant)

cexbrayat commented on docs/fr/api/wrapper-array/at.md:
```diff
@@ -0,0 +1,26 @@
+## at
+
+Retourne `Wrapper` à `index` passé. Utilise une numérotation basée sur les zéros (c'est-à-dire que le premier élément est à l'index 0).
```
Retourne *le* Wrapper?

cexbrayat commented on docs/fr/api/wrapper-array/exists.md:
```diff
@@ -0,0 +1,18 @@
+
+Vérifie que `WrapperArray` existe.
+
+Renvoie false si elle est appelée sur un `WrapperArray` sans objets Wrapper`, ou si l'un d'entre eux n'existe pas.
```
manque une backtick devant Wrapper

cexbrayat commented on docs/fr/api/wrapper/README.md:
```diff
@@ -0,0 +1,57 @@
+
+Vue Test Utils est une API basée sur un wrapper
+
+Un `wrapper`est un objet qui contient un composant ou vnode monté et des méthodes pour tester le composant or vnode
```
manque un espace

cexbrayat replied:
or -> ou

cexbrayat commented on docs/fr/api/wrapper/README.md:
```diff
@@ -0,0 +1,57 @@
+
+### `vm`
+
+`Component`(lecture seule): c'est l'instance `Vue`. Vous pouvez accéder à toutes les [méthodes et propriétés de l'instance d'un vm](https://vuejs.org/v2/api/#Instance-Properties) avec un `wrapper.vm`. Ceci n'existe que pour le wrapper du composant Vue ou le wrapper du composant Vue liant les éléments HTMLE.
```
HTMLE -> HTML

cexbrayat commented on docs/fr/api/wrapper/get.md:
```diff
@@ -0,0 +1,23 @@
+## get
+
+::: warning Avertissement de déprédation
+L'utilisation de `get`pour rechercher un Composant est dépréciée et sera supprimée. Utilisé plutôt [`getCompoenent`](./getComponent.md)
```
manque un espace
Utilisé -> Utilisez
getCompoenent -> getComponent

cexbrayat commented on docs/fr/api/wrapper/isVisible.md:
```diff
@@ -0,0 +1,20 @@
+
+Affirmer que `Wrapper` est visible.
+
+Retourne `false` si un vieil élément a le style `display: none` ou `visibility: hidden`, est situé à l'intérieur de la balise <details> fermée ou possède un attribut caché.
```
vieil?

cexbrayat commented on docs/fr/api/wrapper/setChecked.md:
```diff
@@ -0,0 +1,30 @@
+
+- **Note:**
+
+Lorsque vous essayez de mettre la valeur à state via `v-model` par `radioInput.element.checked = true ; radioInput.trigger('input')`, `v-model` n'est pas déclenché. Le `v-model` est déclenché par l'événement `change`.
```
valeur à state?

cexbrayat commented on docs/fr/api/wrapper/setProps.md:
```diff
@@ -0,0 +1,55 @@
+
+Définir les props de `Wrapper` `vm` et forcer à mettre à jour
+
+Définit les props `Wrapper` `vm` et force la mise à jour.
```
Répétition

cexbrayat commented on docs/fr/api/wrapper/setSelected.md:
```diff
@@ -0,0 +1,26 @@
+
+- **Note:**
+
+Lorsque vous essayez de mettre la valeur à state via `v-model` par `option.element.selected = true ; parentSelect.trigger('input')`, `v-model` n'est pas déclenché. Le `v-model` est déclenché par l'événement `change`.
```
valeur à state ?

cexbrayat commented on docs/fr/guides/common-tips.md:
```diff
@@ -0,0 +1,343 @@
+
+### Écrire des tests asynchrones (nouveau)
+
+Par défaut, les mises à jour des lots de Vue s'effectuent de manière asynchrone (à la prochaine "cloche"). Ceci afin d'éviter les re-rendus DOM inutiles, et les calculs des watchers ([voir les docs](https://vuejs.org/v2/guide/reactivity.html#Async-Update-Queue) pour plus de détails).
```
des lots de Vue?
"cloche" me semble bizarre aussi.

cexbrayat commented on docs/fr/guides/common-tips.md:
```diff
@@ -0,0 +1,343 @@
+
+### Simulation de transition
+
+Bien qu'appeler `await Vue.nextTick()` fonctionne bien pour la plupart des cas d'utilisation, il y a certaines situations où des solutions de contournement supplémentaire sont nécessaires. Ces problèmes seront résolus avant que la bibliothèque `vue-test-utils`ne sorte de la version bêta. Un exemple est celui des composants de test unitaire avec le wrapper `<transition>` fourni par Vue.
```
manque un espace

cexbrayat commented on docs/fr/guides/common-tips.md:
````diff
@@ -0,0 +1,343 @@
+})
+```
+
+Cela remplace le comportement par défaut du composant `<transition>` et rend les enfants dès que la condition booléenne pertinente change, par opposition à l'application de classes CSS, qui est la façon dont le composant `<transition>` de Vue fonctionne.
````
rend -> affiche

cexbrayat commented on docs/fr/guides/dom-events.md:
```diff
@@ -0,0 +1,216 @@
+
+### Les Options
+
+La méthode `trigger` prend en option un objet optionnel `options`. Les propriétés dans cet objet `options` sont ajoutés à l'événement.
```
ajoutés -> ajoutées

cexbrayat commented on docs/fr/guides/testing-async-components.md:
```diff
@@ -0,0 +1,144 @@
+_Vous pouvez en savoir plus sur les mises à jour asynchrones dans la [Vue docs](https://vuejs.org/v2/guide/reactivity.html#Async-Update-Queue)_
+
+En pratique, cela signifie qu'après la mutation d'une propriété réactive, pour affirmer que le changement a été apporté, votre test doit attendre pendant que Vue effectue les mises à jour.
+Un autre moyen est d'utiliser `await Vue.nextTick()`, mais un moyen plus facile et plus propre est de simplement `await`(attendre) la méthode avec laquelle vous avez muté l'état, comme `trigger`
```
manque un espace

cexbrayat commented on docs/fr/guides/testing-async-components.md:
```diff
@@ -0,0 +1,144 @@
+#### Pourquoi ne pas se contenter de `await button.trigger()` ?
+
+Comme expliqué ci-dessus, il y a une différence entre le temps nécessaire à Vue pour mettre à jour ses composants,
+et le temps qu'il faut pour qu'une promesse, comme celle d'`axios` soit tenue.
```
tenue -> résolue

cexbrayat commented on docs/fr/guides/useful-libraries-for-testing.md:
```diff
@@ -0,0 +1,22 @@
+
+[Vue Testing Library](https://github.com/testing-library/vue-testing-library) est un ensemble d'outils visant à tester les composants sans se fier aux détails de la mise en œuvre. Conçu dans un souci d'accessibilité, son approche permet également de remanier facilement les composants.
+
+It is built on top of Vue Test Utils.
```
a supprimer

cexbrayat commented on docs/fr/guides/using-with-typescript.md:
```diff
@@ -0,0 +1,157 @@
+
+> Un exemple de projet pour cette installation est disponible sur [GitHub](https://github.com/vuejs/vue-test-utils-typescript-example).
+
+TypeScript est un sur-ensemble populaire de JavaScript qui ajoute des types et des classes en plus des JS ordinaires. Vue Test Utils inclut les types dans le paquet distribué, donc il fonctionne bien avec TypeScript.
```
des *fonctionnalités* JS ?

phrobinet replied:
J'ai un peu remodelé ma version: 

TypeScript est un superset populaire de JavaScript qui ajoute des types et des classes au dessus du JS normal. Vue Test Utils inclut les types dans le package distribué, il fonctionne donc bien avec TypeScript.

cexbrayat commented on docs/fr/guides/using-with-typescript.md:
````diff
@@ -0,0 +1,157 @@
+$ vue create hello-world
+```
+
+Dans l’invite du CLI, choisissez de `Manually select features`, sélectionnez TypeScript et appuyez sur la touche enter. Cela créera un projet avec TypeScript déjà configuré.
````
de -> l'option

cexbrayat commented on docs/fr/guides/using-with-typescript.md:
```diff
@@ -0,0 +1,157 @@
+Dans l’invite du CLI, choisissez de `Manually select features`, sélectionnez TypeScript et appuyez sur la touche enter. Cela créera un projet avec TypeScript déjà configuré.
+
+::: tip NOTE
+Si vous souhaitez obtenir un guide plus détaillé sur la configuration de Vue avec TypyScript, consultez le [guide de démarrage de Vue avec TypeScript](https://github.com/Microsoft/TypeScript-Vue-Starter).
```
TypyScript -> TypeScript

cexbrayat commented on docs/fr/guides/using-with-typescript.md:
```diff
@@ -0,0 +1,157 @@
+Si vous souhaitez obtenir un guide plus détaillé sur la configuration de Vue avec TypyScript, consultez le [guide de démarrage de Vue avec TypeScript](https://github.com/Microsoft/TypeScript-Vue-Starter).
+:::
+
+L'étape suivante consiste d'ajouter Jest au projet.
```
d'ajouter -> à ajouter

cexbrayat commented on docs/fr/guides/using-with-typescript.md:
```diff
@@ -0,0 +1,157 @@
+
+### Mise en place de Jest
+
+Jest est un outil de test développer par Facebook, visant à fournir une solution de test unitaire sur batterie. Vous pouvez en savoir plus sur Jest en consultant sa [documentation officielle](https://jestjs.io/).
```
développer -> développé
sur batterie -> avec tout le nécessaire?

cexbrayat commented on docs/fr/guides/using-with-typescript.md:
````diff
@@ -0,0 +1,157 @@
+})
+```
+
+C'est tout ce qui faut faire pour que TypeScript et Vue Test Utils travaillent ensemble !
````
qui-> qu'il

cexbrayat commented on docs/fr/guides/using-with-vue-router.md:
````diff
@@ -0,0 +1,83 @@
+wrapper.vm.$route.path // /some/path
+```
+
+### Les gotchas commun
````
gotchas communs -> problèmes courants

cexbrayat commented on docs/fr/guides/using-with-vue-router.md:
```diff
@@ -0,0 +1,83 @@
+
+L'installation de Vue Router ajoute `$route` et `$router` comme propriétés en lecture seule sur le prototype de Vue.
+
+Cela signifie que tous les futurs tests qui tenteront de simuler `$route` et `$router` echoueront
```
echoueront -> échoueront

cexbrayat commented on docs/fr/guides/using-with-vuex.md:
```diff
@@ -0,0 +1,392 @@
+# Utilisation avec Vuex
+
+Dans ce guide, nous verrons comme tester Vuex dans les composants avec Vue Test Utils, et comment aborder le test d'un sore de Vuex.
```
comme -> comment

cexbrayat commented on docs/fr/guides/using-with-vuex.md:
```diff
@@ -0,0 +1,392 @@
+
+Les actions sont des [fonctions fictives de jest](https://jestjs.io/docs/en/mock-functions.html). Ces fonctions fantaisie nous donnent des méthodes pour affirmer si les actions ont été appelées ou non.
+
+Nous pouvons alors affirmer dans nos tests que le talon d'action a été appelé au moment prévu.
```
talon -> fonction fictive?

cexbrayat commented on docs/fr/guides/using-with-vuex.md:
````diff
@@ -0,0 +1,392 @@
+})
+```
+
+Notez que nous utilisons `cloneDeep` pour cloner la configuration du store avant de créer un store avec lui. C'est parce que Vuex fait muter l'option d'options utilisé pour créer le store. Pour s'assurer que nous avons un store propre dans chaque test nous devons cloner l'objet `storeConfig`.
````
répétition d'options

cexbrayat commented on docs/fr/installation/semantic-versioning.md:
```diff
@@ -0,0 +1,3 @@
+## Version sémantique
+
+Vue Test Utils suit [Semantic Versioning](https://semver.org/) dans tous ses projets officiels pour les fonctionnalités et le comportement documentés. Pour les comportementsnon documentés ou les internes exposés, les changements sont décrits dans les [notes de version](https://github.com/vuejs/vue-test-utils/releases).
```
manque un espace
ou les internes  exposés?

cexbrayat commented on docs/fr/installation/using-other-test-runners.md:
```diff
@@ -0,0 +1,60 @@
+
+[Karma](http://karma-runner.github.io/) est un programme de test qui lance des navigateurs, effectue des tests et nous les rapporte.
+
+En plus de Karma, vous pouvez utiliser le cadre [Mocha](https://mochajs.org/) pour écrire les tests, et la bibliothèque [Chai](http://chaijs.com/) pour les assertions des tests. Vous pouvez également consulter [Sinon](http://sinonjs.org/) pour créer des spies et des stubs
```
cadre -> la bibliothèque

cexbrayat commented on docs/fr/installation/using-other-test-runners.md:
```diff
@@ -0,0 +1,60 @@
+
+### Fonctionnement de Vue Test Utils avec mocha-webpack
+
+Une autre stratégie pour tests les SFC consiste à compiler tous nos tests via la webpack et à les exécuter ensuite dans un testeur. L'avantage de cette approche est qu'elle nous donne un support complet pour toutes les fonctionnalités du webpack et du `vue-loader`, donc nous n'avons pas à faire de compromis dans notre code source.
```
la webpack -> webpack
de webpack

cexbrayat commented on docs/fr/installation/using-other-test-runners.md:
manque un espace

cexbrayat replied:
existe.s -> existe.

cexbrayat commented on docs/fr/installation/using-with-jest.md:
```diff
@@ -0,0 +1,119 @@
+## Utilisation des outils de Vue Test Utils avec Jest (recommandé)
+
+Jest est un outils de test développé par Facebook, visant à fournir une solution de test unitaire sur batterie. Vous pouvez en savoir plus sur Jest en consultant sa [documentation officielle](https://jestjs.io/).
```
sur batteries -> avec tout le nécessaire fourni?

cexbrayat commented on docs/fr/installation/using-with-jest.md:
````diff
@@ -0,0 +1,119 @@
+$ npm install --save-dev @vue/test-utils vue-jest
+```
+
+Ensuite, vous devez dire à Jest de transformer les fichiers `.vue` en utilisant `.vu-jest`. Vous pouvez le faire en ajoutant la configuration suivante dans le `package.json` ou dans un [fichier de configuration de Jest](https://jestjs.io/docs/en/configuration) :
````
.vu-jest -> vue-jest

cexbrayat commented on docs/fr/installation/using-with-jest.md:
```diff
@@ -0,0 +1,119 @@
+
+### Couverture du code
+
+Jest peut être utilisé pour générer des rapports de couverture dans plusieur formats. Voici un exemple simple pour commencer :
```
plusieurs

cexbrayat reviewed (commented):
@phrobinet Thanks for your contribution!
I spotted a few typos.

Note: I'm not a huge fan of "affirmation"  and "affirmer" (for "assertion" and "assert" in English I guess). I would use "vérification" and "vérifier".

phrobinet commented:
> 
> 
> @phrobinet Thanks for your contribution!
> I spotted a few typos.
> 
> Note: I'm not a huge fan of "affirmation" and "affirmer" (for "assertion" and "assert" in English I guess). I would use "vérification" and "vérifier".

I'm not a fan of this word either but I couldn't find the right one and "verification" is much better, thanks

phrobinet committed:
Merge branch 'vuejs:dev' into dev

phrobinet committed:
update of the french version in relation to the feedback

lmiller1990 commented:
We can merge once we figure out why Netlify is failing 🤔

lmiller1990 commented:
I see. We need to configure Vuepress to support a new translation. In this file: https://github.com/vuejs/vue-test-utils/blob/dev/docs/.vuepress/config.js

Are you able to do that @phrobinet, or if you don't have time, we can look for someone else to do it (or I can look into it, quite busy fixing 100+ other issues, though.

phrobinet commented:
[configure.md](https://github.com/vuejs/vue-test-utils/files/6740108/configure.md)
This is a configuration of the configure file so that it supports a new translation. Tell me if I need to make any changes to this file?

lmiller1990 commented:
I have no idea how it works, we will just need to play around with it. I'll give it a try now. `yarn vuepress build` fails - need to make it not fail.

lmiller1990 committed:
Merge branch 'dev' into french

lmiller1990 committed:
docs: fix some errors in linting docs

lmiller1990 commented:
Fixed some problems but still getting

wait Rendering static HTML...
[vuepress] No matching page found for sidebar item "/fr/api/components/"
[vuepress] No matching page found for sidebar item "/fr/api/components/"
[vuepress] No matching page found for sidebar item "/fr/api/components/"
[Vue warn]: Error in render: "TypeError: Cannot read property 'match' of undefined"

found in

---> <SidebarLinks>
       <Sidebar>
         <Layout>
           <Layout>
             <GlobalLayout>
               <Root>
[vuepress] No matching page found for sidebar item "/fr/api/components/"
[vuepress] No matching page found for sidebar item "/fr/api/components/"
[vuepress] No matching page found for sidebar item "/fr/api/components/"
[Vue warn]: Error in render: "TypeError: Cannot read property 'match' of undefined"


I don't know why! 🤔

phrobinet commented:
Sorry to answer you so late but I'm back from holiday.

But then I don't understand what is going on 🤨. 
This is the same configuration I use in my vue press projects and it's the same configuration as for the other languages.
I don't see what I've missed on this configuration. 
Maybe you know someone who could answer this question?

phrobinet committed:
Merge branch 'vuejs:dev' into dev

phrobinet committed:
fixed the sidebar and updated the README files in the Wrapper and Installation folders

phrobinet commented:
Hello,
I have understood where the problem with the sidebar was and I have just corrected it. And by the same occasion, I modified two pages (README) which are displayed badly in the French version
Normally it should work

lmiller1990 commented:
Nice job! Just need to run prettier so CI can build, and we can merge. Thanks!

phrobinet committed:
Merge branch 'vuejs:dev' into dev

phrobinet committed:
update with prettier run

lmiller1990 commented:
Merged!! Netlify should auto deploy.
